File: app/src/main/java/com/example/appctruyn/LibraryAdapter.java
package com.example.appctruyn;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.appctruyn.model.LibraryStory;

import java.util.List;

public class LibraryAdapter extends RecyclerView.Adapter<LibraryAdapter.ViewHolder> {

    private final List<LibraryStory> items;
    private final OnItemClickListener onItemClick;
    private final OnMenuClickListener onMenuClick;

    public interface OnItemClickListener {
        void onItemClick(LibraryStory item);
    }

    public interface OnMenuClickListener {
        void onMenuClick(LibraryStory item, View view);
    }

    public LibraryAdapter(List<LibraryStory> items, OnItemClickListener onItemClick, OnMenuClickListener onMenuClick) {
        this.items = items;
        this.onItemClick = onItemClick;
        this.onMenuClick = onMenuClick;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_library, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final ImageView ivCover;
        private final TextView tvTitle;
        private final TextView tvProgress;
        private final ImageView btnNotify;
        private final ImageButton btnMenu;

        public ViewHolder(@NonNull View view) {
            super(view);
            ivCover = view.findViewById(R.id.ivCover);
            tvTitle = view.findViewById(R.id.tvTitle);
            tvProgress = view.findViewById(R.id.tvProgress);
            btnNotify = view.findViewById(R.id.btnNotify);
            btnMenu = view.findViewById(R.id.btnMenu);
        }

        public void bind(final LibraryStory item) {
            tvTitle.setText(item.getTitle());
            
            if (item.getTotalChap() > 0) {
                tvProgress.setText(itemView.getContext().getString(R.string.library_progress, item.getLastChap(), item.getTotalChap()));
            } else {
                tvProgress.setText(itemView.getContext().getString(R.string.library_progress_single, item.getLastChap()));
            }

            btnNotify.setImageResource(
                    item.isNotifyEnabled() ? R.drawable.ic_notify_on : R.drawable.ic_notify_off
            );

            Glide.with(itemView.getContext())
                    .load(item.getCoverUrl())
                    .placeholder(R.drawable.ic_launcher_background)
                    .into(ivCover);

            itemView.setOnClickListener(v -> {
                if (onItemClick != null) {
                    onItemClick.onItemClick(item);
                }
            });

            btnMenu.setOnClickListener(v -> {
                if (onMenuClick != null) {
                    onMenuClick.onMenuClick(item, v);
                }
            });
        }
    }
}

File: app/src/main/java/com/example/appctruyn/LibraryAdapter.kt
package com.example.appctruyn

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.appctruyn.model.LibraryStory

class LibraryAdapter(
    private val items: List<LibraryStory>,
    private val onItemClick: (LibraryStory) -> Unit,
    private val onMenuClick: (LibraryStory, View) -> Unit
) : RecyclerView.Adapter<LibraryAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val ivCover: ImageView = view.findViewById(R.id.ivCover)
        val tvTitle: TextView = view.findViewById(R.id.tvTitle)
        val tvProgress: TextView = view.findViewById(R.id.tvProgress)
        val btnNotify: ImageView = view.findViewById(R.id.btnNotify)
        val btnMenu: ImageButton = view.findViewById(R.id.btnMenu)

        fun bind(item: LibraryStory) {
            tvTitle.text = item.title
            // Hiển thị "Đã đọc X/Y" giống app gốc
            tvProgress.text = if (item.totalChap > 0)
                "Đã đọc ${item.lastChap}/${item.totalChap}"
            else
                "Đã đọc chương ${item.lastChap}"

            // Icon chuông (notify)
            btnNotify.setImageResource(
                if (item.notifyEnabled) R.drawable.ic_notify_on
                else R.drawable.ic_notify_off
            )

            Glide.with(itemView.context)
                .load(item.coverUrl)
                .placeholder(R.drawable.ic_launcher_background)
                .into(ivCover)

            itemView.setOnClickListener { onItemClick(item) }
            btnMenu.setOnClickListener { onMenuClick(item, it) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_library, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount() = items.size
}